### PR TITLE
onboarding connector coding

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

**Comment by DhaaraniCIT** [comment]
Onboarding Connector
Screenshot 2023-10-18 at 2 00 05 PM
Screenshot 2023-10-18 at 2 00 20 PM

File: src/app/core/models/enum/enum.model.ts [hunk 1]
```diff
@@ -117,7 +118,8 @@ export enum ClickEvent {
   QBD_EXPORT = 'Export IIF files',
   CONNECT_INTACCT  = 'Connect Sage Intacct',
   PREVIEW_INTACCT_EXPORT = 'Preview Sage Intacct Export',
-  INTACCT_EXPORT = 'Export Sage Intacct'
+  INTACCT_EXPORT = 'Export Sage Intacct',
+  CONNECT_SAGE300 = 'Connect sage 300 CRE'
```
Comment (ashwin1111) [inline src/app/core/models/enum/enum.model.ts:122]: Sage caps S

File: src/app/core/models/sage300/sage300-db/sage300-credentials.model.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export type Sage300Credential = {
```
Comment (ashwin1111) [inline src/app/core/models/sage300/sage300-db/sage300-credentials.model.ts:1]: folder name can be just db, we are not going to searching with folder name

File: src/app/core/services/sage300/sage300-configuration/sage300-onboarding.service.ts [hunk 3]
```diff
@@ -75,12 +75,12 @@ export class Sage300OnboardingService {
...
   getOnboardingSteps(currentStep: string, onboardingState: Sage300OnboardingState): OnboardingStepper[] {
     this.onboardingSteps.forEach(step => {
-      if (step.step.toLocaleLowerCase() === currentStep.toLocaleLowerCase()) {
+      if (step.step.toLowerCase() === currentStep.toLowerCase()) {
         step.active = true;
       }
     });
 
-    for (let index = this.onboardingStateStepMap[onboardingState]; index > 0; index--) {
+    for (let index = this.onboardingStateStepMap[onboardingState]-1; index > 0; index--) {
```
Comment (ashwin1111) [inline src/app/core/services/sage300/sage300-configuration/sage300-onboarding.service.ts:83]: 1 space before and after -

File: src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.html [hunk 4]
```diff
@@ -0,0 +1,36 @@
+<div>
+    <div>
+       <app-onboarding-steppers [onboardingSteps]="onboardingSteps"></app-onboarding-steppers>
+    </div>
+    <div>
+        <div *ngIf="isLoading" class="tw-flex tw-justify-center tw-items-center tw-h-screen">
+            <app-loader></app-loader>
+        </div>
+        <div *ngIf="!isLoading" class="configuration--contents tw-border-separator tw-mt-6">
```
Comment (ashwin1111) [inline src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.html:9]: configuration css is not found, remove unused class

File: src/app/core/services/sage300/sage300-configuration/sage300-onboarding.service.ts [hunk 5]
```diff
@@ -75,12 +75,12 @@ export class Sage300OnboardingService {
 
   getOnboardingSteps(currentStep: string, onboardingState: Sage300OnboardingState): OnboardingStepper[] {
```
Comment (ashwin1111) [inline src/app/core/services/sage300/sage300-configuration/sage300-onboarding.service.ts:76]: get onboarding state from `this.workspaceService.getOnboardingState()` here, individual components need not send them

File: src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+})
+export class Sage300OnboardingConnectorComponent implements OnInit {
+
+  isLoading: boolean;
+
+  redirectLink = RedirectLink.SAGE300;
+
+  onboardingstep: Sage300OnboardingState = this.workspaceService.getOnboardingState();
+
+  onboardingSteps: OnboardingStepper[] = this.onboardingService.getOnboardingSteps('Connect to Sage 300 CRE', this.onboardingstep);
```
Comment (ashwin1111) [inline src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.ts:24]: can remove 2nd arg

File: src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.ts [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+  templateUrl: './sage300-onboarding-connector.component.html',
+  styleUrls: ['./sage300-onboarding-connector.component.scss']
+})
+export class Sage300OnboardingConnectorComponent implements OnInit {
+
+  isLoading: boolean;
+
+  redirectLink = RedirectLink.SAGE300;
+
+  onboardingstep: Sage300OnboardingState = this.workspaceService.getOnboardingState();
```
Comment (ashwin1111) [inline src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.ts:22]: not needed

File: src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.ts [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+  onboardingSteps: OnboardingStepper[] = this.onboardingService.getOnboardingSteps('Connect to Sage 300 CRE', this.onboardingstep);
+
+  isOnboarding: boolean;
```
Comment (ashwin1111) [inline src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.ts:26]: can remove this, this page will always be in onboarding only

File: src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.ts [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+  constructor(
+    private onboardingService: Sage300OnboardingService,
+    private workspaceService: WorkspaceService,
+    private router: Router,
+    private formBuilder: FormBuilder,
+    private connectorService: Sage300ConnectorService,
+    private toastService: IntegrationsToastService
+  ) { }
+
+  clearField() {
```
Comment (ashwin1111) [inline src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.ts:43]: remove this function, let's keep the form undisturbed incase of failures

File: src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.ts [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+    this.isLoading = true;
+    this.connectorService.connectSage300({
+      user_id: userID,
+      indentifier: companyID,
+      password: userPassword
+    }).subscribe((response) => {
+      this.toastService.displayToastMessage(ToastSeverity.SUCCESS, 'Connection Successful.');
+      this.isLoading = false;
+    }, () => {
+      this.clearField();
```
Comment (ashwin1111) [inline src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.ts:63]: not needed

File: src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.ts [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+    const companyID = this.connectSage300Form.value.companyID;
+    const userPassword = this.connectSage300Form.value.userPassword;
+
+    this.isLoading = true;
+    this.connectorService.connectSage300({
+      user_id: userID,
+      indentifier: companyID,
+      password: userPassword
+    }).subscribe((response) => {
+      this.toastService.displayToastMessage(ToastSeverity.SUCCESS, 'Connection Successful.');
```
Comment (ashwin1111) [inline src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.ts:60]: redirect to export setting post save

File: src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.ts [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+      userID: ['', Validators.required],
+      companyID: ['', Validators.required],
+      userPassword: ['', Validators.required]
+    });
+    this.isLoading = false;
+  }
+
+ngOnInit(): void {
+  this.setupPage();
+}
```
Comment (ashwin1111) [inline src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.ts:82]: indent is off

File: src/app/shared/components/helper/app-landing-page-header/app-landing-page-header.component.ts [hunk 13]
```diff
@@ -71,6 +71,8 @@ export class AppLandingPageHeaderComponent implements OnInit {
       this.trackingService.onClickEvent(ClickEvent.CONNECT_QBD);
     } else if (this.postConnectionRoute==='intacct/onboarding/connector') {
       this.trackingService.onClickEvent(ClickEvent.CONNECT_INTACCT);
+    } else if (this.postConnectionRoute==='sage300/onboarding/connector') {
```
Comment (ashwin1111) [inline src/app/shared/components/helper/app-landing-page-header/app-landing-page-header.component.ts:74]: 1 space before and after ===

File: src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.html (hunk 4 above)
Comment (DhaaraniCIT) [reply src/app/integrations/sage300/sage300-onboarding/sage300-onboarding-connector/sage300-onboarding-connector.component.html:9]: it is there in style.scss file

File: src/app/core/services/sage300/sage300-configuration/sage300-onboarding.service.ts [hunk 14]
```diff
@@ -71,16 +72,20 @@ export class Sage300OnboardingService {
     }
   ];
 
-  constructor() { }
+  onboardingState: Sage300OnboardingState = this.workspaceService.getOnboardingState();
```
Comment (ashwin1111) [inline src/app/core/services/sage300/sage300-configuration/sage300-onboarding.service.ts:75]: private readonly

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**